rama-tcp/src/client/connect.rs

```diff
@@ -23,6 +23,49 @@ use tokio::{
     },
 };
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum IPModes {
```

Comment:
Originally I wanted to go for a single IP Mode indeed, and work with wrapper types. However I think it makes more sense to define 2 enums:

```
enum DnsResolveIpMode
    dual (default)
    singleIpV4
    singleIpV6
    dualPreferIpv4

enum ConnectIpMode
    dual (default)
    ipv4
    ipv6
 ```

Don't think these belong in `rama-tcp` though. Probably more like in `rama-net` or so.

Reply:
Do I need to put these in rama-net?

Reply:
This is still to move to rama-net and `ConnectIpMode` is still to be used as well in the tcp connection part itself (unrelated from DnsResolveIpMode)